In [ ]:
################################### DOXX2TXT
################################### DOCX

In [1]:
import get_file_word as gf
import datetime
fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
fileExt = r".docx"
# NºID,Qtd_caracteres,Qtd_tabela,Qtd_image,data)
            
var1=gf.get_info(fileDir,fileExt,0)
var2=gf.get_info(fileDir,fileExt,1)
var3=gf.get_info(fileDir,fileExt,2)

print(var1)
print(var2)
print(var3)


#gf.get_info(fileDir,fileExt)


True
True
True
('233867', 59270, 3, 11, datetime.date(2019, 12, 6))
('233885', 99723, 19, 9, datetime.date(2020, 2, 5))
('233886', 73923, 18, 23, datetime.date(2020, 2, 5))


In [3]:
print(var1[4])


2019-12-06


In [ ]:
# Importar bibliotecas
from os import listdir
from os.path import isfile, join
import docx2txt
import docx
import os
import glob

from os import listdir
from os.path import isfile, join

In [ ]:
import glob
#targetPattern = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/*.docx"
targetPattern = r"/Users/walnerpessoa/repositorios/Fluxo_Editorial/words/*.docx"

#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
print(targetPattern)
files_array = glob.glob(targetPattern)
print(files_array)

In [ ]:
#fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
fileExt = r".docx"
[_ for _ in os.listdir(fileDir) if _.endswith(fileExt)]

In [ ]:
# definir uma variável para o local onde estão os arquivos Word
#mypath="/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'

# gerar vetor com ons nomes dos arquivos Word
onlyfiles = [f for f in listdir(fileDir) if isfile(join(fileDir, f))]

# verificar se existe o diretório
#os.path.exists('/Users/wpessoa/repositorios/Fluxo_Editorial/words')
os.path.exists(fileDir) 

In [ ]:
print(onlyfiles[0])
#docx2txt.process(onlyfiles[1])
doc = docx.Document(fileDir+'/'+onlyfiles[0])
#print(fileDir+'/'+onlyfiles[0])


In [ ]:
indice = 0
image_array=[]
table_chunks = []

# abrir conecção com o arquivo Word 
doc = docx.Document(fileDir+'/'+onlyfiles[indice])

# data criação
dt_doc = doc.core_properties.created
# ler em cada parágrafo dentro do arquivo Word
paragra= [p.text for p in doc.paragraphs]

# extai a quantidade de caracteres
caracteres = docx2txt.process(fileDir+'/'+onlyfiles[indice])

# gerar um array com tamanho de todas as imagens
for image in doc.inline_shapes:
    image_array.append([image.width, image.height])
    # print (image.width, image.height)
    
# gerar um array com todas as tabelas
for table in doc.tables:
    table_chunks.append(table)

#Saída dos dados
print('Nome do diretório: ',fileDir)
print('Nome do arquivo: ',onlyfiles[indice])
print("Data de Criação do Arquivo: ", dt_doc,"\n")
print('Quantidade de carcteres:',len(caracteres))
print("Quantidade de parágrafos: ", len(paragra))
print("Quantidade de imagens no vetor de imagens/gráficos: ",len(image_array))
print("Quantidade de Tabelas: ",len(table_chunks))


Nome do diretório:  /Users/wpessoa/repositorios/Fluxo_Editorial/words/
Nome do arquivo:  ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx
Quantidade de carcteres: 59270

Quantidade de parágrafos:  356
Quantidade de imagens no vetor de imagens:  11
Quantidade de Tabelas:  3
Data de Criação do Arquivo:  2019-12-06 13:09:00

In [ ]:
#####

In [ ]:
#teste com metodos da biblioteca Docx

print(doc.paragraphs[0].runs)
print(doc.paragraphs[0].text)
print(doc.paragraphs[0].runs[0].text)
print(doc.paragraphs[0].runs[0].style)
#print(doc.paragraphs[1].runs[0].text)
#print(doc.paragraphs[1].runs[0].style)
print(doc.core_properties.created)
print(doc.inline_shapes)
print(doc.inline_shapes[0].type)
print(doc.inline_shapes[0].height)
print(doc.inline_shapes[0].width)

#
